src/components/signup/SignupState.js

```diff
@@ -153,14 +235,17 @@ const Signup = ({ navigation, screenProps }: { navigation: any, screenProps: any
 
       await Promise.all([
         addUserAPIPromise,
-        userStorage.setProfile({ ...state, walletAddress: goodWallet.account }),
+        userStorage.setProfile({ ...requestPayload, walletAddress: goodWallet.account }),
         userStorage.setProfileField('registered', true, 'public'),
         goodWallet.getBlockNumber().then(creationBlock => userStorage.saveLastBlockNumber(creationBlock.toString())),
       ])
 
+      await AsyncStorage.removeItem('web3Token')
+      await API.updateW3UserWithWallet(requestPayload.w3Token, goodWallet.account)
```

Comment:
we dont need to await for that, since we dont do anything with the response
add .catch and log error

Reply:
done